js/ad_manager_videoplaza.js

```diff
@@ -464,56 +420,36 @@ OO.Ads.manager(function(_, $) {
       OO.log("ad found! " + ad);
       // When the ad is done, trigger callback
       var ui = this.amc.ui;
-      var streamUrl = ad.ad.streamUrl;
+
       this.lastVideoAd = null;
 
-      //There is a caching bug with chrome that this will fix.
-      if (OO.isChrome) {
-        streamUrl = streamUrl + (/\?/.test(streamUrl) ? "&" : "?") + "_=" + OO.getRandomString();
-      }
       if (ad.isLinear) {
-        var innerWrapper = this.amc.ui.videoWrapper;
-        var onTimeUpdate = _.bind(function (ad, adPodEndedCallback, streamUrl, event) {
-          // Originally [jigish] wrote This is a hack fix for m3u8, current iOS has a bug that if the m3u8 EXTINF indication
-          // a different duration, the ended event never got dispatched. Monkey patch here to manual trigger a onEnded event
-          // need to wait OTS to fix their end. [gfrank] Doesn't seem to be fixed so using it here.
-          var duration = ad.duration;
-          var durationInt = Math.floor(duration);
-          var isM3u8 = streamUrl.toLowerCase().indexOf("m3u8") > 0;
-          this.trackQuartiles(ad, duration, this.amc.ui.adVideoElement[0].currentTime);
-          if (isM3u8 && this.amc.ui.adVideoElement[0].currentTime >= duration && duration > durationInt) {
-            this.amc.ui.adVideoElement.off("timeupdate", onTimeUpdate);
-            this.amc.ui.adVideoElement.off("ended", onEnded);
-            _.delay(_.bind(this.adCompleted, this, adPodEndedCallback), 0, event);
-          }
-        }, this, ad, adPodEndedCallback, streamUrl);
-        var onEnded = _.bind(function (ad, adPodEndedCallback, event) {
-          this.amc.ui.adVideoElement.off("ended", onEnded);
-          this.amc.ui.adVideoElement.off("timeupdate", onTimeUpdate);
-          this.adCompleted(adPodEndedCallback);
-        }, this, ad, adPodEndedCallback);
-        ui.adVideoElement.on("ended", onEnded);
-        ui.adVideoElement.on("timeupdate", onTimeUpdate);
-        ui.adVideoElement[0].src = streamUrl;
-        ui.adVideoElement[0].load(false);
-        var widthOfPlayer = innerWrapper.width();
-        var heightOfPlayer = innerWrapper.height();
-        ui.adVideoElement.css({
-          "display": "block", "width": widthOfPlayer, "height": heightOfPlayer,
-          "text-align": "center"
-        });
-        this.checkCompanionAds(ad.ad);
-        this.quartileTracks = {};
-        ui.adVideoElement[0].play();
         this.lastVideoAd = ad;
-        this.tracker.trackEvent(ad.ad.creatives[0], videoplaza.tracking.Tracker.CreativeEventType.START);
-        adPodStartedCallback();
-        ui.rootElement.find('div.oo_tap_panel').css('display', 'block');
+        this.amc.notifyPodStarted(ad.id, 1);
+        adCompletedCallback = _.bind(function(amc, adId) {
+          amc.notifyLinearAdEnded(adId);
+          amc.notifyPodEnded(adId);
```

Comment:
Would it prove beneficial to have a function to encompass the functionality of notifying of a linear ad pod end? There is another instance of these two functions in cancelAd.